tests/integration/targets/azure_rm_privatednszonelink/tasks/main.yml

```diff
@@ -0,0 +1,127 @@
+- name: Create virtual network link name
+  set_fact:
+    link_name: "link{{ resource_group | hash('md5') | truncate(22, True, '') }}"
+
+- name: Create random number
+  set_fact:
+    rpfx: "{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}"
+  run_once: yes
+
+- name: Create random domain name
+  set_fact:
+    domain_name: "{{ resource_group | hash('md5') | truncate(16, True, '') + (65535 | random | string) }}"
+
+- name: Create a Private DNS zone
+  azure_rm_privatednszone:
+    resource_group: "{{ resource_group }}"
+    name: "{{ domain_name }}.com"
+    state: present
+
+- name: Create a virtual network
+  azure_rm_virtualnetwork:
+    name: "vnet{{ rpfx }}"
+    resource_group: "{{ resource_group }}"
+    address_prefixes_cidr:
+        - 10.1.0.0/16
+        - 172.100.0.0/16
+    dns_servers:
+        - 127.0.0.1
+        - 127.0.0.2
+
+- name: Create a subnet
+  azure_rm_subnet:
+    name: "subnet{{ rpfx }}"
+    virtual_network_name: "vnet{{ rpfx }}"
+    resource_group: "{{ resource_group }}"
+    address_prefix_cidr: 10.1.0.0/24
+
+- name: Create a virtual network link
+  azure_rm_privatednszonelink:
+    resource_group: "{{ resource_group }}"
+    name: "{{ link_name }}"
+    zone_name: "{{ domain_name }}.com"
+    virtual_network: "vnet{{ rpfx }}"
+    state: present
+  register: results
+
+- name: Assert that virtual network link is created
+  assert:
+    that: results.changed
+
+- name: Create a virtual network link (Idempotent test)
+  azure_rm_privatednszonelink:
+    resource_group: "{{ resource_group }}"
+    name: "{{ link_name }}"
+    zone_name: "{{ domain_name }}.com"
+    virtual_network: "vnet{{ rpfx }}"
+    state: present
+  register: results
+
+- name: Assert that output is not changed
+  assert:
+    that: not results.changed
+
+- name: Update a virtual network link
+  azure_rm_privatednszonelink:
+    resource_group: "{{ resource_group }}"
+    name: "{{ link_name }}"
+    zone_name: "{{ domain_name }}.com"
+    registration_enabled: true
+    virtual_network: "vnet{{ rpfx }}"
+    state: present
+  register: results
+
+- name: Assert that virtual network link is updated
+  assert:
+    that: results.changed
+
+- name: Get virtual network link
+  azure_rm_privatednszonelink_info:
+    resource_group: "{{ resource_group }}"
+    name: "{{ link_name }}"
+    zone_name: "{{ domain_name }}.com"
+  register: results
+
+- assert:
+    that:
+      - not results.changed
+      - results.virtualnetworklinks[0].name == "{{ link_name }}"
+      - results.virtualnetworklinks[0].registration_enabled == true
+      - results.virtualnetworklinks[0].provisioning_state == "Succeeded"
+      - results.virtualnetworklinks[0].virtual_network_link_state == "Completed"
```

Comment:
It is recommended to delete this parameter, because due to environmental reasons, the return value of this parameter does not respond immediately, sometimes the value is (InProgress or Complete)
```suggestion

```